tests/job/test_job_interactive.py

```diff
@@ -0,0 +1,22 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import unittest
+from pyiron_base.job.interactive import InteractiveBase
+from pyiron_base._tests import TestWithProject
+
+
+class TestJobInteractive(TestWithProject):
+    def test_job_with(self):
+        job = self.project.create_job(InteractiveBase, "job_modal")
+        with self.assertRaises(NotImplementedError):
+            with job as _:
+                pass
+                
+    def test_job_interactive_with(self):
+        job = self.project.create_job(InteractiveBase, "job_interactive")
+        job.project.db.add_item_dict(job.db_entry())
+        with job.interactive_open() as job_int:
+            job_int.to_hdf()
+        self.assertTrue(job.server.run_mode.interactive)
```

Comment:
```suggestion
        self.assertFalse(job.server.run_mode.interactive, msg="Job failed to interactively close on exiting with")
```

Right?

Reply:
`run_mode` doesn't change so what @jan-janssen wrote is correct.

Reply:
Wait, interactive close leaves the run mode as interactive? That seems bad

Reply:
Well, as long as `job.status` is set to `finished` I think it's fine.

Reply:
The `run_mode` queue also remains when the job is finished, just like `non_modal`. Basically the run mode is only internal for pyiron to understand how to execute the job. The option that changes is `job.interactive_is_activated()` but this should never be touched by the user. As we only test the `wrapper` without any explicit interface to an interactive job we can not test this option.